File: src/main/java/com/fwdekker/randomness/array/ArraySettingsAction.java
package com.fwdekker.randomness.array;

import com.fwdekker.randomness.SettingsAction;
import com.fwdekker.randomness.SettingsDialog;


/**
 * Controller for random array generation settings.
 */
public final class ArraySettingsAction extends SettingsAction {
    @Override
    public SettingsDialog createDialog() {
        return new ArraySettingsDialog();
    }

    @Override
    public String getTitle() {
        return "Array Settings";
    }
}

File: src/main/kotlin/com/fwdekker/randomness/array/ArraySettingsAction.kt
package com.fwdekker.randomness.array

import com.fwdekker.randomness.DataQuickSwitchSchemeAction
import com.fwdekker.randomness.DataSettingsAction
import icons.RandomnessIcons


/**
 * Controller for random array generation settings.
 *
 * @see ArraySettings
 * @see ArraySettingsComponent
 */
class ArraySettingsAction : DataSettingsAction() {
    override val name = "Array Settings"

    override val configurableClass = ArraySettingsConfigurable::class.java


    /**
     * Opens a popup to allow the user to quickly switch to the selected scheme.
     *
     * @param settings the settings containing the schemes that can be switched between
     */
    class ArrayQuickSwitchSchemeAction(settings: ArraySettings = ArraySettings.default) :
        DataQuickSwitchSchemeAction<ArrayScheme>(settings, RandomnessIcons.Data.QuickSwitchScheme) {
        override val name = "Quick Switch Array Scheme"
    }
}
